requests/sessions.py

```diff
@@ -203,22 +203,26 @@ def resolve_redirects(self, resp, req, stream=False, timeout=None,
             # Override the original request.
             req = prepared_request
 
-            resp = self.send(
-                req,
-                stream=stream,
-                timeout=timeout,
-                verify=verify,
-                cert=cert,
-                proxies=proxies,
-                allow_redirects=False,
-                **adapter_kwargs
-            )
+            if yield_requests:
+                yield req
```

Comment:
Ok, so I just stumbled across this in the code, and this is *very* confusing. Despite this being a generator, it is a generator that can literally be iterated only once in this mode before it starts doing something stupid (where "stupid" is that it repeatedly runs the same processing). That's not actually written down anywhere, so it would probably help us to at least say that the "yield requests" mode can only yield *one* request.